lektor_s3.py

```diff
@@ -33,6 +36,16 @@ def __init__(self, env, output_path):
         self.bucket = None
         self.key_prefix = ''
         self.cloudfront = None
+        self.config = self.get_config()
+
+    @property
+    def config_filename(self):
+        return os.path.join(self.env.root_path, 'configs', 's3.ini')
+
+    def get_config(self):
```

Comment:
The `Plugin` class comes with a `get_config` method that does this logic already. Loading the config should be the work of the `S3Plugin`; it can pass the config object in to `S3Publisher`'s `__init__`.

Two reasons this is worth doing:
  - it's good to be in sync with the lektor project in case they move stuff
  - it will clean up a lot of the monkeypatching in tests

I think you might get a slightly different api than what `ConfigParser` gives you - you'll get a mitsuhiko-style INI (https://github.com/mitsuhiko/python-inifile/blob/master/inifile.py).